### ISSUE TITLE
Add snes.smp.canonical support

### ISSUE THREAD

**Issue by KungFuFurby** [ARM9/bass#30]
Until bass v11 (citing the history kit for me tracing this down), the program had C++-side code that supported the canonical SPC700 syntax. This was never ported over to the .arch format (which meant it was only available as raw C++ programming), and thus has been cut from the assembler since v11.

**Comment by thar0x29a** [comment]
Hi, yes you are right. It never had been ported even thought its possible to do so. If someone would like to port it, I would love to add such an pull request back into the main project. 
But right now, since its just a 'flavor' thing, I will most likely not do it. 
Because even if I had finished all the cool features that I planed to do: There just too many other CPU Architectures out there, that we do not support right now. And I reater would add a new one, then do cosmetic's. 

In other words: Feel free to add them. Im sure, that with some spirit it should be not too difficult to write an convertor that ports the current arch files to the original SPC700 syntax.

**Comment by pokun** [comment]
> Until bass v11 (citing the history kit for me tracing this down), the program had C++-side code that supported the canonical SPC700 syntax. This was never ported over to the .arch format (which meant it was only available as raw C++ programming), and thus has been cut from the assembler since v11.
> 
> I'm interested in this mainly for stylistic reasons, as surprisingly I found that this reminded me of the Z80's native syntax once I went to analyze some of the Z80's assembly.

I made an arch file for the official SPC700 syntax a few years ago. It can be downloaded at the bottom of the linked forum post here: https://forums.nesdev.org/viewtopic.php?p=262655#p262655 ("bass_spcn.7z")

**Comment by thar0x29a** [comment]
> > Until bass v11 (citing the history kit for me tracing this down), the program had C++-side code that supported the canonical SPC700 syntax. This was never ported over to the .arch format (which meant it was only available as raw C++ programming), and thus has been cut from the assembler since v11.
> > I'm interested in this mainly for stylistic reasons, as surprisingly I found that this reminded me of the Z80's native syntax once I went to analyze some of the Z80's assembly.
> 
> I made an arch file for the official SPC700 syntax a few years ago. It can be downloaded at the bottom of the linked forum post here: https://forums.nesdev.org/viewtopic.php?p=262655#p262655 ("bass_spcn.7z")

Oh, cool! I will have a look at it and if everything looks nice, I should add it to our dev branch
